test/integration/testpaths/testpath_hypervisor_limit.py

```diff
@@ -0,0 +1,198 @@
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        self.cleanup = []
+        if self.skiptest:
+            self.skipTest("This test is to be checked on xenserver \
+                    only  Hence, skip for %s" % self.hypervisor)
+
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+
+    def tearDown(self):
+        try:
+
+            cmd = updateHypervisorCapabilities.updateHypervisorCapabilitiesCmd()
+            cmd.id = self.hostCapId
+            cmd.maxguestslimit = self.originalLimit
+            self.apiclient.updateHypervisorCapabilities(cmd)
+
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["advanced", "basic"], required_hardware="false")
+    def test_check_hypervisor_max_limit_effect(self):
+        """ Test hypervisor max limits effect
+
+        # 1. Read exsiting count of VM's on the host 
+                and modify maxguestlimit of the host by exsiting count + 4
+        # 2. Deploy a VM
+        # 2. Try to deploy another vm
+        # 3. Verify that second VM
+                deployment fails (2 SSVMs 1 VR VM and 1 deployed VM)
+        """
+
+        # Step 1
+        listVm = list_virtual_machines(self.apiclient,
+                                       hostid=self.host.id)
+        vmCount = len(listVm)
+        # Increament vm count by 4(2 SSVMs 1 VR VM and 1 deployed VM)
+        newValue = vmCount + 4
+
+        qresultset = self.dbclient.execute(
+            "select hypervisor_version from host where uuid='%s'" %
+            self.host.id)
+
+        event_validation_result = validateList(qresultset)
+        self.assertEqual(
+            event_validation_result[0],
+            PASS,
+            "event list validation failed due to %s" %
+            event_validation_result[2])
+
+        cmdList = listHypervisorCapabilities.listHypervisorCapabilitiesCmd()
+        cmdList.hypervisor = self.hypervisor
+        config = self.apiclient.listHypervisorCapabilities(cmdList)
+
+        for host in config:
+            if host.hypervisorversion == qresultset[0][0]:
+                self.hostCapId = host.id
+                self.originalLimit = host.maxguestslimit
+                break
+        else:
+            self.skipTest("No hypervisor capabilities found for %s \
+                    with version %s" % (self.hypervisor, qresultset[0][0]))
+
+        cmdUpdate = updateHypervisorCapabilities.\
+            updateHypervisorCapabilitiesCmd()
+        cmdUpdate.id = self.hostCapId
+        cmdUpdate.maxguestslimit = newValue
+        self.apiclient.updateHypervisorCapabilities(cmdUpdate)
```

Comment:
Here you are updating the Hypervisor Capabilities with new value with in our e.g. is 7. So in this case how you are expecting that the 2nd vm deployment would fail?